### PR TITLE
feat: support schema change by idx and reverse

### PR DESCRIPTION
**Pull request by fansehep**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-rs/issues/4908.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-schema/src/schema.rs [hunk 1]
```diff
@@ -57,6 +57,17 @@ impl SchemaBuilder {
         self.fields.remove(idx)
     }
 
+    /// Change the FieldRef as index `idx`
+    /// if index out of bounds, will panic
+    pub fn change_field(&mut self, idx: usize, field: impl Into<FieldRef>) {
+        self.fields[idx] = field.into()
```
Comment (tustvold) [inline arrow-schema/src/schema.rs:63]:
```suggestion
    /// Returns a mutable reference to the field at `idx`
    pub fn field_mut(&mut self, idx: usize) -> &mut FieldRef {
        &mut self.fields[idx]
```
What do you think about this instead?

Comment (fansehep) [reply arrow-schema/src/schema.rs]: I think it's good!!! 😄

**Comment by tustvold** [comment]
Thank you

**Comment by fansehep** [comment]
> Thank you

Thans for your review. 😍

**Pull request merged**